java/dds/dcps_java.mpc

```diff
@@ -82,13 +83,22 @@ project: idl2jni, javah, dcpslib, optional_jni_check, dcps_java_optional, dcps_t
     OpenDDS/DCPS/transport/UnableToCreateException.java
   }
 
+  verbatim (gnuace, top, 1) {
+    opendds_android_network_callback = $(if $(android_target_api)>23,AndroidNetworkCallback.java,FakeAndroidNetworkCallback.java)
```

Comment:
So I haven't thought about this before, but I don't think GNU make can't do integer comparisons. `$(if)` just checks the string length of the condition.

```make
all:
	echo $(if 1>3,ERROR,Good)
```

Results in `ERROR`

I think the only real option is use something like:

```make
has_android_network_callback ?=
ifdef android_target_api
  ifeq ($(shell expr $(android_target_api) '>=' 24),1)
    has_android_network_callback = 1
  endif
endif
all:
	echo $(if $(has_android_network_callback),Yes,No)
```

```
$ make -f test_makefile android_target_api=23
No
$ make -f test_makefile android_target_api=24
Yes
```